### PR TITLE
chore: common model for Salesforce identifiers

### PR DESCRIPTION
**Pull request by zregvart**
Eliminates the necessity of a mapping step between Salesforce connectors
by using a common `SalesforceIdentifier` DTO from `salesforce-model`.

### CODE REVIEW

File: connectors/twitter-mention-connector/src/main/java/io/syndesis/connector/twitter/springboot/TwitterMentionConnectorConfigurationCommon.java [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package io.syndesis.connector.twitter.springboot;
 
 import javax.annotation.Generated;
+import io.syndesis.connector.twitter.TwitterMentionComponent;
```
Comment (igarashitm) [inline connectors/twitter-mention-connector/src/main/java/io/syndesis/connector/twitter/springboot/TwitterMentionConnectorConfigurationCommon.java]: It looks this import just came alone without use?

**Review by igarashitm — COMMENTED** [review]
Only one trivial comment

File: connectors/twitter-mention-connector/src/main/java/io/syndesis/connector/twitter/springboot/TwitterMentionConnectorConfigurationCommon.java (hunk 1 above)
Comment (rhuss) [inline connectors/twitter-mention-connector/src/main/java/io/syndesis/connector/twitter/springboot/TwitterMentionConnectorConfigurationCommon.java]: Why is this import needed ?

File: connectors/salesforce-model/src/main/java/io/syndesis/connector/salesforce/SalesforceIdentifier.java [hunk 2]
```diff
@@ -18,13 +18,13 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
-public class AnyObject {
+public class SalesforceIdentifier {
```
Comment (rhuss) [inline connectors/salesforce-model/src/main/java/io/syndesis/connector/salesforce/SalesforceIdentifier.java:21]: Sorry got a bit confused about all those files (which are cut in GitHub, too) and I'm a bit lazy to check this out locally :). But I _thought_ that the SalesforceIdentifer comes from the mode. Why is it defined in this connector, too ?

**Review by rhuss — COMMENTED** [review]
LGTM on a brief view, but have two comments.

File: connectors/twitter-mention-connector/src/main/java/io/syndesis/connector/twitter/springboot/TwitterMentionConnectorConfigurationCommon.java (hunk 1 above)
Comment (zregvart) [reply connectors/twitter-mention-connector/src/main/java/io/syndesis/connector/twitter/springboot/TwitterMentionConnectorConfigurationCommon.java]: yeah, those are generated by camel-package-maven-plugin (IIRC), I'll modify it locally and update

File: connectors/salesforce-model/src/main/java/io/syndesis/connector/salesforce/SalesforceIdentifier.java (hunk 2 above)
Comment (zregvart) [reply connectors/salesforce-model/src/main/java/io/syndesis/connector/salesforce/SalesforceIdentifier.java:21]: yeah I was lazy and moved AnyObject from one of the connectors into `salesforce-model` that's why you see rename here, there is only one, don't trust me, trust find:

```shell
[zregvart@zregvart connectors]$ find . -name SalesforceIdentifier.java
./connectors/salesforce-model/src/main/java/io/syndesis/connector/salesforce/SalesforceIdentifier.java
[zregvart@zregvart connectors]$ 
```

Comment (rhuss) [reply connectors/salesforce-model/src/main/java/io/syndesis/connector/salesforce/SalesforceIdentifier.java:21]: sorry, thought it was still in the delete connector. all good :)

**Review by rhuss — APPROVED** [review]

**Comment by rhuss** [comment]
@zregvart fine with a new release ? 0.6.0 maybe ?

**Comment by zregvart** [comment]
> @zregvart fine with a new release ? 0.6.0 maybe ?

@rhuss yes, thank you 👍
